src/OpenRiaServices.Hosting.AspNetCore/Test/OpenRiaServices.Hosting.AspNetCore.Test/OperationInvokerTests.cs

```diff
@@ -0,0 +1,201 @@
+    {
+        var operation = domainServiceDescription.GetQueryMethod(nameof(OperationCanceledDomainService.Query_CancelAndAbort));
+        
+        var operationInvoker = new QueryOperationInvoker<City>(operation, GetSerializationHelper());
+        
+        await InvokeAndAssertNoResponseIsWritten(operationInvoker);
+    }
+
+    [TestMethod]
+    [Description("Submit operation: Cancellation is requested but method returned succesfully")]
+    public async Task TestSubmit_CancelAndReturn()
+    {
+        var submit = new ReflectionDomainServiceDescriptionProvider.ReflectionDomainOperationEntry(domainServiceDescription.DomainServiceType,
+            typeof(DomainService).GetMethod(nameof(DomainService.SubmitAsync)), DomainOperation.Custom);
+
+        var operationInvoker = new SubmitOperationInvoker(submit, GetSerializationHelper());
+
+        await InvokeAndAssertNoResponseIsWritten(operationInvoker, SubmitType.Cancel);
+    }
+
+    [TestMethod]
+    [Description("Submit operation: Cancellation is requested and method throws OperationCanceledException")]
+    public async Task TestSubmit_CancelAndAbort()
+    {
+        var submit = new ReflectionDomainServiceDescriptionProvider.ReflectionDomainOperationEntry(domainServiceDescription.DomainServiceType,
+            typeof(DomainService).GetMethod(nameof(DomainService.SubmitAsync)), DomainOperation.Custom);
+
+        var operationInvoker = new SubmitOperationInvoker(submit, GetSerializationHelper());
+
+        await InvokeAndAssertNoResponseIsWritten(operationInvoker, SubmitType.CancelAndThrow);
+    }
+
+    private static SerializationHelper GetSerializationHelper() => new (domainServiceDescription);
+
+    private static async Task InvokeAndAssertNoResponseIsWritten(OperationInvoker operationInvoker)
+    {
+        var context = GetHttpContext();
+        await operationInvoker.Invoke(context);
+        Assert.IsTrue(context.Response.ContentLength is null, "A response should not have been written");
+    }
+
+    private static async Task InvokeAndAssertNoResponseIsWritten(SubmitOperationInvoker operationInvoker, SubmitType submitType)
+    {
+        var bytes = GetEmptyChangeSet(Array.Empty<ChangeSetEntry>());
+        var context = GetHttpContext(submitType);
+        context.Request.Method = "POST";
+        context.Request.Body = bytes;
+        await operationInvoker.Invoke(context);
+        Assert.IsTrue(context.Response.ContentLength is null, "A response should not have been written");
+    }
+
+    private static HttpContext GetHttpContext(SubmitType submitType = SubmitType.Cancel)
+    {
+        var httpContext = new DefaultHttpContext();
+        httpContext.Response.Body = new MemoryStream();
+
+        var cts = new CancellationTokenSource();
+        var domainService = new OperationCanceledDomainService(cts, submitType);
+
+        var services = new ServiceCollection();
+        services.AddScoped(typeof(OperationCanceledDomainService));
```

Comment:
I think you can remove this lince now that it is registered below
```suggestion
```

Reply:
Applied the suggstion, verified it worked localy